src/controller/UserRegistrationController.ts

```diff
@@ -33,45 +39,109 @@ export class UserRegistrationController {
    *                 example: senha#67@Maluca!
    *     responses:
    *       201:
-   *         description: JSON with user data creates new user successfully
+   *         description: Only status without body
+   *       500:
+   *         description: Only status without body
+   */
+  public static async userRegistration(req: Request, res: Response): Promise<void> {
+    try {
+      const { id, name, email, password } = req.body;
+
+      const confirmationToken = await JwtUtils.generateJWTToken({ userId: id }, '24h');
+
+      const newUser: DeepPartial<User> = {
+        name,
+        email,
+        password
+      };
+
+      newUser.confirmationToken = confirmationToken;
+      newUser.pendingConfirmation = true;
+
+      await UserRepository.createUser(newUser);
+
+      NodemailerService.sendUserRegistrationConfirmationEmail(email);
```

Comment:
essa função não é assíncrona? se ela for chamada sem o await, corre risco do endpoint finalizar e não ter mandado o e-mail para o usuário, é bom já realizar o teste da funcionalidade e adicionar o await para garantir que esteja funcionando

Reply:
Feito